clang/test/CheckedCRewriter/b30_structprotocastimplicitunsafeuseunsafe.c

```diff
@@ -34,12 +34,10 @@ struct r {
 
 
 struct r *sus(struct r *, struct r *);
-	//CHECK_NOALL: struct r *sus(struct r *x : itype(_Ptr<struct r>), struct r *y : itype(_Ptr<struct r>)) : itype(_Ptr<struct r>);
-	//CHECK_ALL: struct r *sus(struct r *x : itype(_Ptr<struct r>), struct r *y : itype(_Ptr<struct r>));
+	//CHECK: struct r *sus(struct r *x : itype(_Ptr<struct r>), struct r *y : itype(_Ptr<struct r>)) : itype(_Ptr<struct r>);
 
 struct r *foo() {
-	//CHECK_NOALL: _Ptr<struct r> foo(void) {
-	//CHECK_ALL: struct r *foo(void) {
+	//CHECK: _Ptr<struct r> foo(void) {
```

Comment:
Nice. What change resulted in this fix? Something to do with how pointers were being made WILD, perhaps?

Reply:
`getInvalidCastPVCons` was always storing the generated constraint variable in the `ExprConstraintVars` map. This happened even when the cast being processed was an implicit cast. Since implicit casts don't appear in the source, we shouldn't map a `PersistentSourceLocation` to the constraint. 

When that does happen, there's a chance that it collides with another expression, replacing the constraint for the expression with a WILD variable.